misc/python/materialize/scratch.py

```diff
@@ -226,22 +226,29 @@ def is_ready(i: Instance) -> bool:
     mkrepo(i, git_rev)
 
 
-def mkrepo(i: Instance, rev: str) -> None:
-    mssh(i, "git init --bare materialize/.git")
+def mkrepo(i: Instance, rev: str, init: bool = True, force: bool = False) -> None:
+    if init:
+        mssh(i, "git init --bare materialize/.git")
+
+    rev = git.rev_parse(rev)
+
+    cmd: List[str] = [
+        "git",
+        "push",
+        "--no-verify",
+        f"{instance_host(i)}:materialize/.git",
+        # Explicit refspec is required if the host repository is in detached
+        # HEAD mode.
+        f"{rev}:refs/heads/scratch",
+    ]
+    if force:
```

Comment:
optional nit: I think it's safe to always pass `--force` here too

Reply:
I always pass it in `push`, ill leave `create` the same for now